jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/AbstractRuntimeManager.java

```diff
@@ -431,49 +440,28 @@ protected void createLockOnGetEngine(Context<?> context, RuntimeEngine runtime)
     }
     
     protected void createLockOnGetEngine(Long id, RuntimeEngine runtime) {
-        if (!isUseLocking()) {
-            logger.debug("Locking on runtime manager disabled");
+        if (id == null) {
             return;
         }
-        
-        if (id != null) {
-            ReentrantLock newLock = new ReentrantLock();
-            ReentrantLock lock = engineLocks.putIfAbsent(id, newLock);
-            if (lock == null) {
-                lock = newLock;
-                logger.debug("New lock created as it did not exist before");
-            } else {
-                logger.debug("Lock exists with {} waiting threads", lock.getQueueLength());
-            }
-            logger.debug("Trying to get a lock {} for {} by {}", lock, id, runtime);
-            lock.lock();
-            logger.debug("Lock {} taken for {} by {} for waiting threads by {}", lock, id, runtime, lock.hasQueuedThreads());
-            
+        try {
+            runtimeManagerLockStrategy.lock(id, runtime);
+        } catch(InterruptedException e) {
+            throw new RuntimeException("Runtime manager was not able to lock for {" + id +"}", e);
```

Comment:
Style, show we just throw a more specifc runtime here? For example, IllegalStateException

Reply:
I didn't create any specific actually and it is not really a problem with the state.. should be something like RuntimeManagerLockException but it is quite cumbersome...